### PR TITLE
 feat(hsdo): add support for multiple backend configuration

### PR DESCRIPTION
**Pull request by christian-vdz**
## Context
We need HSDO to be able to add servers in multiple HAProxy backends.

## How?
By editing the code so that we can provide a list of backends.

### CODE REVIEW

**Review by chubchubsancho — APPROVED** [review]

File: src/client/haproxy.py [hunk 1]
```diff
@@ -17,10 +17,19 @@ def __init__(self):
...
+        # self.backendList = {
+        #     "usvc-origin": {
+        #         "baseName": "usvc-origin",
+        #         "serverPort": "80"
+        #     },
+        #     "manifest-proxy": {
+        #         "baseName": "manifest-proxy",
+        #         "serverPort": "8080"
+        #     }
+        # }
```
Comment (vgnx) [inline src/client/haproxy.py]: Why adding commented lines ?

**Review by vgnx — APPROVED** [review]
OK, but I think commented code can be removed if it's not used

File: src/client/haproxy.py (hunk 1 above)
Comment (christian-vdz) [reply src/client/haproxy.py]: It's just to give an example to make review easier

Comment (yverry) [reply src/client/haproxy.py]: ok, say it in comment

**Review by yverry — APPROVED** [review]

File: src/client/haproxy.py (hunk 1 above)
Comment (christian-vdz) [reply src/client/haproxy.py]: done

**Review by shlagevuk — APPROVED** [review]

File: src/client/haproxy.py [hunk 2]
```diff
@@ -45,22 +55,26 @@ def checkBackendConf(self):
...
         for backend in stat:
             values = backend.split(",")
             ## Example backend line
             ## http_back,mywebapp2,0,0,0,0,,0,0,0,,0,,0,0,0,0,UP,10,1,0,0,1,330587,0,,1,4,2,,0,,2,0,,0,L4OK,,0,0,0,0,0,0,0,,,,,0,0,,,,,-1,,,0,0,0,0,,,,Layer4 check passed,,2,3,4,,,,10.14.34.198:80,,http,,,,,,,,0,0,0,,,0,,0,0,0,0,
-            if len(values) > 80 and values[0] == self.backendName and values[1].startswith(self.backendBaseName):
-                backendExists = True
-            elif self.azLimiter == "true" and len(values) > 80 and values[0] == self.fallbackBackendName and values[1].startswith(self.fallbackBackendBaseName):
-                fallbackBackendExists = True
+            for backend in backendList:
+                if len(values) > 80 and values[0] == backend and values[1].startswith(backendList[backend]["baseName"]):
```
Comment (gforien) [inline src/client/haproxy.py]:
```suggestion
            for k,v in backendList.items():
                if len(values) > 80 and values[0] == k and values[1].startswith(v["baseName"]):
```
you can iterate over a dictionnary with `.items()`

Comment (gforien) [inline src/client/haproxy.py]:
```suggestion
                if len(values) < 80:
                    continue
                if values[0] == backend and values[1].startswith(backendList[backend]["baseName"]):
```
If you always need `len(values) > 80`, you can put it above

File: src/client/haproxy.py [hunk 3]
```diff
@@ -45,22 +55,26 @@ def checkBackendConf(self):
...
+                    existingBackends += 1
+                elif self.azLimiter == "true" and len(values) > 80 and values[0] == self.fallbackBackendName and values[1].startswith(self.fallbackBackendBaseName):
+                    fallbackBackendExists = True
```
Comment (gforien) [inline src/client/haproxy.py]:
```suggestion
                elif self.azLimiter == "true" and len(values) > 80 and values[0] == self.fallbackBackendName and values[1].startswith(self.fallbackBackendBaseName):
                    fallbackBackendExists = True
```
these two lines do not depend of variable `backend`, so you can move this statement out of the `for` loop

**Review by valentinpelus — APPROVED** [review]

**Pull request merged**